src/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation/build/netstandard2.0/Microsoft.AspNetCore.Mvc.Razor.ViewCompilation.targets

```diff
@@ -58,15 +58,15 @@
   <Target
     Name="_AddDesktopReferences"
     AfterTargets="ResolveLockFileReferences"
-    Condition="'$(_MvcViewCompilationAddDesktopReferences)'!='false' AND '$(TargetFrameworkIdentifier)'=='.NETFramework'">
+    Condition="'$(MvcRazorCompileOnPublish)'=='true' AND '$(_MvcViewCompilationAddDesktopReferences)'!='false' AND '$(TargetFrameworkIdentifier)'=='.NETFramework'">
 
     <PropertyGroup Condition="'$(_MvcViewCompilationBinaryPath)'==''">
       <_MvcViewCompilationBinaryPath Condition="'$(PlatformTarget)'=='x64'">$(MSBuildThisFileDirectory)$(MSBuildThisFileName)-x64.exe</_MvcViewCompilationBinaryPath>
       <_MvcViewCompilationBinaryPath Condition="'$(PlatformTarget)'=='x86'">$(MSBuildThisFileDirectory)$(MSBuildThisFileName)-x86.exe</_MvcViewCompilationBinaryPath>
     </PropertyGroup>
 
     <ItemGroup Condition="'$(_MvcViewCompilationBinaryPath)'!=''">
-      <Reference Include="$(_MvcViewCompilationBinaryPath)" Private="true" />
+      <Reference Include="$(_MvcViewCompilationBinaryPath)" Private="false" Visible="false" />
```

Comment:
See https://msdn.microsoft.com/en-us/library/bb629388.aspx - we don't want to copy local the binary. It ends up in the deps file (which worked in the functional tests, but blows up in MusicStore)